token/program-2022/src/processor.rs

```diff
@@ -1244,6 +1246,13 @@ impl Processor {
         accounts: &[AccountInfo],
         new_extension_types: Vec<ExtensionType>,
     ) -> ProgramResult {
+        if new_extension_types
+            .iter()
+            .any(|&t| t == ExtensionType::MintCloseAuthority || t == ExtensionType::Uninitialized || t == ExtensionType::ConfidentialTransferMint)
+        {
+            return Err(TokenError::ExtensionTypeMismatch.into());
+        }
```

Comment:
This only covers two of the mint extensions. To make this more robust, you can change this to
```suggestion
        if new_extension_types
            .iter()
            .any(|&t| t.get_account_type() != AccountType::Account)
        {
            return Err(TokenError::ExtensionTypeMismatch.into());
        }
```

Using the specific helper for this https://github.com/solana-labs/solana-program-library/blob/5900d4d85f3662dd910abe84bafe0656ae4ba495/token/program-2022/src/extension/mod.rs#L930